# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../zthread/bugfix.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./include/zthread/Barrier.h.orig	Sun Mar 13 06:23:00 2005
+++ ./include/zthread/Barrier.h	Sat Jun 21 17:31:14 2008
@@ -150,7 +150,7 @@
 
       } else {
 
-        int myGeneration = _generation;
+        unsigned int myGeneration = _generation;
 
         try {
 
@@ -251,7 +251,7 @@
 
       } else {
 
-        int myGeneration = _generation;
+        unsigned int myGeneration = _generation;
 
         try {
 
--- ./include/zthread/BlockingQueue.h.orig	Sun Mar 13 03:04:18 2005
+++ ./include/zthread/BlockingQueue.h	Sat Jun 21 17:31:14 2008
@@ -83,7 +83,7 @@
       /**
        * @see Queue::add(const T& item, unsigned long timeout)
        */
-      virtual bool add(T item, unsigned long timeout) {
+      virtual bool add(const T& item, unsigned long timeout) {
 
         try {
 
